Split the "ASoC: SOF: Intel: use sof_sdw as default SDW machine driver" commit

bardliao opened the pull request:
To extract the skip checking dummy topology part from the "ASoC: SOF: Intel: use sof_sdw as default SDW machine driver" commit. So that we can submit the "ASoC: SOF: don't check the existence of dummy topology" commit first and use a dummy topology in the acpi match table. And no need to create an unused topology,
No code change in this PR.

bardliao committed:
ASoC: SOF: Intel: use sof_sdw as default SDW machine driver

If there is no SoundWire machine matches the existing acpi match table,
get the required machine data from the acpi table and construct the
link adrs and endpoints. Pass the data to the default Intel SoundWire
machine driver. And we don't need to add new item to the acpi match
table in common cases.
We will construct a dummy topology name. The dummy topology is just
used to extract the platform name for function topology and should not
be used.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

bardliao committed:
Revert "ASoC: SOF: Intel: use sof_sdw as default SDW machine driver"

This reverts commit 5f38e117c959b6786757c1705962a40254687930.
To split the commit into 2 commits. One to skip checking dummy topology
and the other to use sof_sdw as default SDW machine driver.

bardliao committed:
ASoC: SOF: don't check the existence of dummy topology

Monolithic topology is not needed if function topology is used. Instead
of creating a dummy topology in the file system, we can skip the
existence check in the kernel.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

ujfalusi commented on sound/soc/sof/topology.c:
```diff
@@ -2523,6 +2523,11 @@ int snd_sof_load_topology(struct snd_soc_component *scomp, const char *file)
 	if (!tplg_cnt) {
+		if (strstr(file, "dummy")) {
+			dev_err(scomp->dev,
+				"Function topology is required, please upgrade sof-bin\n");
```
in other fail paths we refer to 'sof-firmware' package, we should use it here as well.

bardliao replied:
Thanks, updated.

ujfalusi reviewed (commented):
@bard only on thing to change: s/sof-bin/sof-firmware for the error path print to align with other code paths.

bardliao committed:
fixup! ASoC: SOF: Intel: use sof_sdw as default SDW machine driver

We can't use default SDW machine driver if links[link_index].adr_d is
null. Also, add a paranoid check to avoid kernel NULL pointer
dereference.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

ujfalusi commented on sound/soc/sof/intel/hda.c:
```diff
@@ -1391,6 +1400,8 @@ static struct snd_soc_acpi_mach *hda_sdw_machine_select(struct snd_sof_dev *sdev
 		links[link_index].adr_d = find_acpi_adr_device(sdev->dev, peripherals->array[i],
 							       &links[link_index], &amp_index);
+		if (!links[link_index].adr_d)
+			return NULL;
```
This is an error? Should we print a message?

bardliao replied:
Yes, this is an error, and the error message is printed in find_acpi_adr_device().

ujfalusi commented on sound/soc/sof/intel/hda.c:
```diff
@@ -1184,6 +1184,15 @@ static struct snd_soc_acpi_adr_device *find_acpi_adr_device(struct device *dev,
+		if (!name_prefix) {
+			dev_err(dev, "codec_info_list name_prefix of part id %#x is missing\n",
+				codec_info_list[i].part_id);
+			return NULL;
```
I take this has happened, but how is that possible?

bardliao replied:
In case if someone add new codecs in codec_info_list[], but doesn't add a name_prefix for it. And the null pointer will happen when `if (!strcmp(name_prefix, "Left"))` is called.
